tutorial/markdown/java/spring-boot.md

```diff
@@ -53,172 +91,494 @@ mvn package
+7. The response is sent back to the client with the updated airline data if found, or a 404 Not Found response if the airline with the specified ID does not exist.
+
+### DELETE Mapping Workflow
+
+`@DeleteMapping("/{id}")`
+
+The DELETE mapping is triggered when a client sends an HTTP DELETE request to `/api/v1/airline/{id}`, where `{id}` is the unique identifier of the airline to be deleted.
+
+1. The `AirlineController` receives the request and extracts the `id` from the URL path.
+2. The `AirlineController` calls the `deleteAirline` method of the `AirlineService`, passing the `id` of the airline to be deleted. This function internally calls [airlineCol.remove()](https://docs.couchbase.com/java-sdk/current/howtos/kv-operations.html#removing) to remove the airline from the database.
+3. The `AirlineService` is responsible for deleting the airline from the database using the `AirlineRepository`.
+4. If the airline is deleted successfully, the `AirlineService` performs the deletion operation without returning any response data.
+5. The `AirlineController` constructs an HTTP response with a status code of 204 No Content, indicating that the request was successful, but there is no content to return in the response body.
+6. The response is sent back to the client to indicate the successful deletion of the airline.
+
+These workflows illustrate how each HTTP method interacts with the `AirlineService` and the underlying database through the `AirlineRepository` to perform various operations on airline data.
+
+## Custom SQL++ Queries
+
+1. Get all airlines by country
 
 ```java
-profileCol.upsert(id, profile);
-return ResponseEntity.status(HttpStatus.CREATED).body(profile);
+
+    @Override
+    public List<Airline> findByCountry(String country, int limit, int offset) {
+      String statement = "SELECT airline.id, airline.type, airline.name, airline.iata, airline.icao, airline.callsign, airline.country FROM `"
+                + dbProperties.getBucketName() + "`.`inventory`.`airline` WHERE country = '" + country + "' LIMIT "
+                + limit + " OFFSET " + offset;
+        return cluster
+                .query(statement, QueryOptions.queryOptions().scanConsistency(QueryScanConsistency.REQUEST_PLUS)
+                        .parameters(JsonObject.create().put("country", country)))
+                .rowsAs(Airline.class);
+
+    }
 ```
 
-> *from update method of controllers/ProfileController.java*
+<!-- Explaination -->
+
+In the above example, we are using the `QueryOptions` class to set the `scanConsistency` to `REQUEST_PLUS` to ensure that the query returns the latest data. We are also using the `JsonObject` class to set the `country` parameter in the query. [Query Options and Scan Consistency](https://docs.couchbase.com/java-sdk/current/howtos/n1ql-queries-with-sdk.html#scan-consistency).
```

Comment:
The link should not stand out. It should be part of the explanation (example `QueryOptions`) for people to refer if they have doubts.

Reply:
added it to part of explaination